### PR TITLE
feat: let-else postfix

### PR DESCRIPTION
**Pull request by saiintbrisson**
Creates a new `lete` postfix standing for the recent `let-else` addition.

The postfix follows the rules for `ifl/match` and completes with Ok/Some
if the expression is a try enum.

2023-11-15 19-46-40 (online-video-cutter com)

### CODE REVIEW

File: crates/ide-completion/src/completions/postfix.rs [hunk 1]
```diff
@@ -186,6 +200,13 @@ pub(crate) fn complete_postfix(
                 &format!("match {receiver_text} {{\n    ${{1:_}} => {{$0}},\n}}"),
             )
             .add_to(acc, ctx.db);
+
+            postfix_snippet(
+                "lete",
+                "let pat = expr else {}",
+                &format!("let $1 = {receiver_text} else {{\n    $0\n}};"),
+            )
+            .add_to(acc, ctx.db);
```
Comment (Young-Flash) [inline crates/ide-completion/src/completions/postfix.rs:209]: I think it's a irrefutable `let...else` pattern, like `let foo = 0 else { return; }`

Comment (saiintbrisson) [reply crates/ide-completion/src/completions/postfix.rs:209]: One is able to match against any pattern they wish, like match, 
`let Enum::Variant(val)` = ... else ...`

**Comment by Alainx277** [comment]
I have a pull request implementing this feature that has been stuck 😕 
https://github.com/rust-lang/rust-analyzer/pull/15730

**Comment by Veykril** [comment]
Apologies for that, I've been on vacation for 3 weeks and only came back a week ago so I have to slowly catch up with the PRs again.

**Comment by saiintbrisson** [comment]
@Alainx277 thank you for the heads-up. I don't know how I missed it!

**Pull request closed without merging**